Desktop/engine/enginesync.cpp

```diff
@@ -514,26 +519,89 @@ void EngineSync::processReloadData()
 }
 
 
-bool EngineSync::processScriptQueue()
+stringset EngineSync::processRScriptQueue()
 {
+	bool	foundEngine		= false, 
+			engineNotIdle	= false;
+	
 	try
 	{
-		for(auto * engine : _engines)
+		if(_waitingScripts.size() > 0)
 		{
-			if(engine->idle()  && engine->runsUtility() && _waitingScripts.size() > 0)
+			RScriptStore * waiting = _waitingScripts.front();
+			
+			if(waiting->typeScript == engineState::rCode)
 			{
-				RScriptStore * waiting = _waitingScripts.front();
-
-				switch(waiting->typeScript)
+				const std::string mod = fq(waiting->module);
+			
+				if(!moduleHasEngine(mod))	
+				{
+					for(auto & engine : _engines)
+						if(engine->idle() && engine->module() == "")
+						{
+							registerEngineForModule(engine, mod);	
+							foundEngine		= true;
+							engineNotIdle	= true; //it still needs to load the module
+							
+							if(!engine->moduleLoaded() && !engine->moduleLoading())
+								engine->moduleLoad();
```

Comment:
Why the registerEngineForModule does not care that the moduleLoad function is called?

Reply:
because it just "registers" that this engine and module belong together, but that could also be used for installing a module. well, how would you load a module you've yet to install? ;)